homeassistant/components/mqtt/quality_scale.yaml

```diff
@@ -0,0 +1,122 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup:
+    status: done
+    comment: >
+      Entities are updated through dispatchers, and these are
+      cleaned up when the integration unloads.
+  entity-unique-id:
+    status: exempt
+    comment: >
+      This is user configurable, but not required.
+      It is required though when a user wants to use device based discovery.
+  has-entity-name: done
+  runtime-data:
+    status: exempt
+    comment: >
+      Runtime data is not used, as the mqtt entry data is only used to set up the
+      MQTT broker, this happens during integration setup,
+      and only one config entry is allowed.
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable: done
+  entity-unavailable:
+    status: done
+    comment: |
+      Only supported for entities the user has assigned a unique_id.
+  action-exceptions: done
+  reauthentication-flow: done
+  parallel-updates: done
+  test-coverage: done
+  integration-owner: done
+  docs-installation-parameters: done
+  docs-configuration-parameters: done
+
+  # Gold
+  entity-translations:
+    status: exempt
+    comment: >
+      This is not possible because the integrations generates entities
+      based on a user supplied config or discovery.
+  entity-device-class:
+    status: done
+    comment: An entity device class can be configured by the user for each entity.
+  devices:
+    status: done
+    comment: >
+      A device context can be configured by the user for each entity.
+      It is not required though, except when using device based discovery.
+  entity-category:
+    status: done
+    comment: An entity category can be configured by the user for each entity.
+  entity-disabled-by-default:
+    status: done
+    comment: >
+      The user can configure this through YAML or discover
+      entities that are disabled by default.
+  discovery:
+    status: done
+    comment: >
+      When the Mosquitto MQTT broker add on is installed,
+      a MQTT config flow allows an automatic setup from its discovered settings.
+  stale-devices:
+    status: exempt
+    comment: >
+      This is is only supported for entities that are configured through MQTT discovery.
+      Users must manually cleanup stale entities that were set up though YAML.
+  diagnostics: done
+  exception-translations: done
+  icon-translations:
+    status: exempt
+    comment: >
+      This is not possible because the integrations generates entities
+      based on a user supplied config or discovery.
+  reconfiguration-flow: done
```

Comment:
Is this supported? I don't see a `async_step_reconfigure` so are there other ways to reconfigure because mqtt is special or would this be considered exempt?

Reply:
May be, but all settings can be configured, But you are right step_reconfigure is not used.

Reply:
This is how the UI displays the re-configuration button:
![afbeelding](https://github.com/user-attachments/assets/e18356df-f647-46a4-b9af-e458c50c0bda)

The is different, as it has some additional tooling on the Config entry settings page.

Reply:
OK thanks. The context for all of these is trying to write programmatic checks in #131413 to enforce them, but my approach may be too simplistic. Appreciate the detail.

Reply:
It is possible to add a reconfiguration flow for MQTT, but then then the part behind the option flow would be redundant.